server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMessagesMethod.java

```diff
@@ -90,73 +93,83 @@
         Preconditions.checkArgument(request instanceof GetMessagesRequest);
         GetMessagesRequest getMessagesRequest = (GetMessagesRequest) request;
         Optional<ImmutableSet<MessageProperty>> requestedProperties = getMessagesRequest.getProperties();
+        Optional<ImmutableSet<MessageHeaderProperty>> headerProperties = getMessagesRequest.getHeaderProperties();
         return Stream.of(JmapResponse.builder().clientId(clientId)
                             .response(getMessagesResponse(mailboxSession, getMessagesRequest, requestedProperties))
                             .responseName(RESPONSE_NAME)
-                            .properties(requestedProperties.map(this::handleSpecificProperties))
-                            .filterProvider(Optional.of(buildFilteringHeadersFilterProvider(requestedProperties)))
+                            .properties(handleSpecificProperties(requestedProperties, headerProperties))
```

Comment:
this little change makes nested calls a mess, can't we manage this "isPresent()" more gracefully ?